KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
+ref14_3 = []
+
+# specref9 is the KLS indexes that go in chapter 9
 specref9 = []
+
+# specref14 is the KLS indexes that in chapter 14
 specref14 = []
-comms = ""  # holds the ACTUAL STRINGS of the commands
+
+check = []
+check2 = []
+check3 = []
+kls_list = []
+sortmatch = []
+
+w, h = 3, 9999
+sorterIIIcheck = [[0 for x in range(w)] for y in range(h)]
+
+sortmatch_2 = []
+
+test = "memesbananashgvlksrjc%/n"
+print test[-2]
 # 2/18/16 this method addresses the goal of hardcoding in the necessary packages to let the chapter files run as pdf's.
-# Currently only works with chapter 9, ask Dr. Cohl to help port your
-# chapter 14 output file into a pdf
+# Currently only works with chapter 9, ask Dr. Cohl to help port your chapter 14 output file into a pdf
+
+# 5/5/16 this method implements the Hypergeometric paragraphs into relevant sections for practicality
+# Later versions should attempt to: fix paragraphs with variations on this name like
+# Basic Hypergeometric Representation, q-Hypergeometric Representation
+
+
+def new_keywords(kls):
+    """
+    This section checks through the Addendum and identifies words that are valid keywords of sections
+    It then takes that list and removes duplicates, as well as printing the output to another file.
+
+    :param kls: The addendum that provides the words.
+    :return: Inputs results into a global list.
+    """
+
+    kls_list_chap = []
+    for i in kls:
+        if kls.index(i) > 313:
+            if "paragraph{" in i or "subsubsection*{" in i:
+                kls_list.append(i[i.find("{") + 1: len(i) - 2])
+
+    i = 0
+    while i < len(kls_list):
+        if "reference" in kls_list[i].lower() or "limit relation" in kls_list[i].lower():
+
+            del kls_list[i]
+        else:
+            i += 1
+    kls_list.append("Limit Relation")
+    for i in kls_list:
+        add = True
+        for j in kls_list_chap:
+            if i == j:
+                add = False
+        if add:
+            kls_list_chap.append(i)
+    return kls_list_chap
 
 
-def prepareForPDF(chap):
-    footmiscIndex = 0
+def fix_chapter_sort(kls, chap, word, sortloc):
+    """
+    This function sorts through the input files and identifies and places sections from the addendum after their
+    respective subsections in the chapter.
+
+    :param kls: The addendum where the sections to be inserted are found.
+    :param chap: The destination of inserted sections.
+    :param word: The keyword that is being processed.
+    :param sortloc: The location in the list of words, of the word being sorted.
+    :return: This function outputs the processed chapter into a larger function for further processing.
+    """
+    hyper_headers_chap = []
+    hyper_subs_chap = []
+
+    sep1 = 0
+    name_chap = word.lower()
+    if name_chap == "orthogonality":
+        sep1 = 1
+    elif name_chap == "special value":
+        sep1 = 2
+
+    khyper_header_chap = []
+    k_hyper_sub_chap = []
+    index = 0
+
+    for i in kls:
```

Comment:
Oh and actually in case of something as large as this loop, it is probably better to use a more descriptive name than `item` too, maybe something that actually shows what the variable is. See http://stackoverflow.com/a/209065/1718670 for why good variable naming practices are valuable. See also: http://stackoverflow.com/a/209089/1718670